sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/TryCast.scala

```diff
@@ -84,4 +84,14 @@ case class TryCast(child: Expression, dataType: DataType, timeZoneId: Option[Str
 
   override def typeCheckFailureMessage: String =
     AnsiCast.typeCheckFailureMessage(child.dataType, dataType, None, None)
+
+  override def toString: String = {
+    s"try_cast($child as ${dataType.simpleString})"
+  }
+
+  override def sql: String = dataType match {
+    case _: ArrayType | _: MapType | _: StructType => child.sql
```

Comment:
can we copy the original comment as well?

Reply:
Talked to @cloud-fan offline. Actually, we don't need this legacy special handling.